### PR TITLE
[cxx-interop] Add support for C++ comparison operators.

### PR DESCRIPTION
**Pull request by zoecarver**
Adds support for C++ operators: `<`, `>`, `==`, `!=`, `<=`, and `>=`.

### CODE REVIEW

**Comment by zoecarver** [comment]
@swift-ci please smoke test

File: lib/ClangImporter/ImportName.cpp [hunk 1]
```diff
@@ -1427,6 +1427,12 @@ ImportedName NameImporter::importNameImpl(const clang::NamedDecl *D,
     case clang::OverloadedOperatorKind::OO_GreaterGreater:
     case clang::OverloadedOperatorKind::OO_AmpAmp:
     case clang::OverloadedOperatorKind::OO_PipePipe:
+    case clang::OverloadedOperatorKind::OO_Less:
+    case clang::OverloadedOperatorKind::OO_Greater:
+    case clang::OverloadedOperatorKind::OO_EqualEqual:
+    case clang::OverloadedOperatorKind::OO_ExclaimEqual:
+    case clang::OverloadedOperatorKind::OO_LessEqual:
+    case clang::OverloadedOperatorKind::OO_GreaterEqual:
```
Comment (MForster) [inline lib/ClangImporter/ImportName.cpp:1435]: Can we keep this list (and also the test cases) in the same order as https://github.com/llvm/llvm-project/blob/master/clang/include/clang/Basic/OperatorKinds.def#L58,L104?

**Review by gribozavr — APPROVED** [review]

File: lib/ClangImporter/ImportName.cpp (hunk 1 above)
Comment (zoecarver) [reply lib/ClangImporter/ImportName.cpp:1436]: Good idea. I did this as a separate commit.

**Comment by zoecarver** [comment]
@swift-ci please smoke test and merge.

**Comment by zoecarver** [comment]
@swift-ci please smoke test and merge.

**Comment by zoecarver** [comment]
@swift-ci please smoke test and merge.

**Pull request merged**